### PR TITLE
Update contribution doc

### PR DESCRIPTION
**Pull request by shashwathi**
### What this PR does / why we need it

Update contribution doc

### CODE REVIEW

**Review by odinnordico — APPROVED** [review]

**Review by warango4 — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,73 +1,70 @@
...
-When your branch gets out of sync with the vmware-tanzu/main branch, use the following to update:
+### Development Environment Setup
 
-``` shell
-git checkout my-new-feature
-git fetch -a
-git pull --rebase upstream main
-git push --force-with-lease origin my-new-feature
-```
+Follow the documentation in the development document for the [pre-requisites](./Development.md#Prerequisites) required for the setup, as well [building](./Development.md#building) the project.
```
Comment (atmandhol) [inline CONTRIBUTING.md:25]: pre-reqs and building links should be capital `./DEVELOPMENT.md`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,73 +1,70 @@
...
-If you need to squash changes into an earlier commit, you can use:
+## Contribution Flow
 
-``` shell
-git add .
-git commit --fixup <commit>
-git rebase -i --autosquash main
-git push --force-with-lease origin my-new-feature
-```
+This project operates according to the talk, then code rule. If you plan to submit a pull request for anything more than a typo or obvious bug fix, first you should [raise an issue][new-issue] to discuss your proposal, before submitting any code.
```
Comment (atmandhol) [inline CONTRIBUTING.md:36]: [raise an issue] not linked properly

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,73 +1,70 @@
...
+Before submitting your pull request, we advise you to use the following:
 
-We follow the conventions on [How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/).
+### Pull Request Checklist
 
-Be sure to include any related GitHub issue references in the commit message.  See
-[GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues
-and commits.
+1. Check if your code changes will pass both code linting checks and unit tests.
+2. Ensure your commit messages are descriptive. We follow the conventions on [How to Write a Git Commit Message](http://chris.beams.io/posts/git-commit/). Be sure to include any related GitHub issue references in the commit message. See [GFM syntax](https://guides.github.com/features/mastering-markdown/#GitHub-flavored-markdown) for referencing issues and commits.
```
Comment (atmandhol) [inline CONTRIBUTING.md:54]: I would change this bullet point with out conventional commit standard as follows.

Commit messages should follow the following format:
```
type: <short-summary>
```

Add ! after type if it's a breaking change
```
type!: <short-summary>
```

**NOTE**: `short-summary` should explain what the commit does. The automation tools we build will basically take this `short-summary` and put it in a changelog. So write your `short-summary` in a way that is worthy to go in the changelog.

`type`: can be one of the following:
- `build`: Changes that affect the build system (example: Makefile)
- `chore`: Housekeeping tasks, grunt tasks etc; no production code change
- `ci`: Changes to our CI configuration files and scripts
- `docs`: Documentation only changes
- `deprecate`: A code change that deprecates a feature
- `feat`: A new feature
- `fix`: A bug fix
- `perf`: A code change that improves performance
- `refactor`: A code change that neither fixes a bug nor adds a feature, includes updates to a feature.
- `removed`: Removed a feature that was previously deprecated
- `test`: Adding missing tests or correcting existing tests
```

**Comment by shashwathi** [comment]
@atmandhol : I have addressed all your comments. Please review when you get a chance again

File: CONTRIBUTING.md (hunk 1 above)
Comment (atmandhol) [reply CONTRIBUTING.md]: This still seems to be pointing to a page that does not exists.

`./Development.md#Prerequisites` should be `./DEVELOPMENT.md#Prerequisites`
`./Development.md#building` should be `./DEVELOPMENT.md#building`

Comment (shashwathi) [reply CONTRIBUTING.md]: Good to know. Surprisingly in Visual studio code, links worked. Looks like github links is strongly typed. Learnt something new :)

**Review by atmandhol — APPROVED** [review]

**Review by rashedkvm — APPROVED** [review]

**Pull request merged**